modern-cpp-font-lock.el

```diff
@@ -53,85 +53,104 @@
   "Provides font-locking as a Minor Mode for Modern C++"
   :group 'faces)
 
-(defun modern-c++-string-lenght< (a b) (< (length a) (length b)))
-(defun modern-c++-string-lenght> (a b) (not (modern-c++-string-lenght< a b)))
-
-(defcustom modern-c++-types
-  (eval-when-compile (sort '("bool" "char" "char16_t" "char32_t" "double" "float" "int" "long" "short" "signed" "unsigned" "void" "wchar_t")
-                           'modern-c++-string-lenght>))
+(eval-and-compile
+  (defun modern-c++-string-lenght< (a b) (< (length a) (length b)))
+  (defun modern-c++-string-lenght> (a b) (not (modern-c++-string-lenght< a b))))
+
```

Comment:
Note: when I look at the `elc` compiled file, defvar symbols appears. They disappear if the `--` variables are placed directly in the `defcustom`. In both cases, values are (now correctly) compiled time generated.